dracut/30ignition/ignition-generator

```diff
@@ -41,9 +41,36 @@ if $(cmdline_bool 'ignition.firstboot' 0); then
 
     # Invoke distro hook for detecting whether we're booted from a live image,
     # and therefore won't have a root disk.
+    diskful=
     if ! command -v is-live-image >/dev/null || ! is-live-image; then
         add_requires ignition-diskful.target
+        diskful=1
     fi
+
+    # ignition-setup-user.service should depend on the boot device node
+    # only on diskful boots
+    add_requires ignition-setup-user.service
+    cat > "${UNIT_DIR}/ignition-setup-user.service" <<EOF
```

Comment:
Going back to dynamically spitting out whole units is unfortunate...

Hmm, WDYT about keeping `ignition-setup-user.service` statically enabled and just make the generator add a drop-in in the `diskful` case so it waits on the boot device?

Reply:
I did think about this too, but didn't feel super strong about it.